src/Dictionaries/HashedDictionaryCollectionTraits.h

```diff
@@ -0,0 +1,74 @@
+#pragma once
+
+#include <type_traits>
+#include <sparsehash/sparse_hash_map>
+#include <Common/HashTable/Hash.h>
+
+namespace DB
+{
+
+/// sparse_hash_map/sparse_hash_set
+template <typename C>
+concept IsSparseHashTable = std::is_same_v<C, google::sparse_hash_map<
+    typename C::key_type,
+    typename C::mapped_type,
+    /// HashFcn is not exported in sparse_hash_map is public type
+    DefaultHash<typename C::key_type>>>;
+template <typename V>
+concept IsStdMapCell = requires (V v)
+{
+    v->first;
+    v->second;
+};
+
+/// HashMap/HashSet and their Cells
+template <typename C>
+concept IsBuiltinHashTable = !IsSparseHashTable<C>;
```

Comment:
I guess the concept will break if someone adds another external hash table in future. Maybe `IsBuildinHashTable` should check directly against the building hash table type.

Reply:
I doubt that it worth it, since we will need too much code (almost 30 lines), but let's try.